### PR TITLE
manifests/infrastructure: Setup the platform specific status fields.

### PR DESCRIPTION
**Pull request by abhinavdahiya**
the API were added in https://github.com/openshift/api/pull/300 and https://github.com/openshift/api/pull/359

Platform specific status fields allow installer a way to provide details about the platform to the operators.

/cc @jhixson74 @patrickdillon 
^^-- for the changes
/cc@coreydaley @ironcladlou 
^^-- for openshift-ingress and openshift-image-registry operator API for GCP.

Closes https://github.com/openshift/installer/pull/1725

### CODE REVIEW

**Comment by ironcladlou** [comment]
Awesome, thanks!

/lgtm

**Comment by ironcladlou** [comment]
@abhinavdahiya Well, I didn't realize I was an approver here when I tagged this and now it merged without wide review — sorry.

**Comment by dmage** [comment]
/cc @coreydaley 
(in the description there is no space before "@", so I guess GitHub didn't send a notification)

File: pkg/asset/manifests/infrastructure.go [hunk 1]
```diff
@@ -82,13 +66,50 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
...
+	case openstack.Name:
+		config.Status.Platform = configv1.OpenStackPlatformType
+		config.Status.PlatformStatus.Type = configv1.OpenStackPlatformType
+	case vsphere.Name:
+		config.Status.Platform = configv1.VSpherePlatformType
+		config.Status.PlatformStatus.Type = configv1.VSpherePlatformType
+	default:
+		config.Status.PlatformStatus.Type = configv1.NonePlatformType
+	}
+	config.Status.Platform = config.Status.PlatformStatus.Type
```
Comment (patrickdillon) [inline pkg/asset/manifests/infrastructure.go]: It seems pointless to have this assignment and the same assignment in the case statements. The assignment in the case statements will be undone by this. Which one we keep depends on whether they are always equal or if there can be skew depending on the platform.

Comment (ironcladlou) [reply pkg/asset/manifests/infrastructure.go]: Good catch, agree.

Comment (abhinavdahiya) [reply pkg/asset/manifests/infrastructure.go]: opened https://github.com/openshift/installer/pull/1934 to fix it.

File: pkg/asset/manifests/infrastructure.go [hunk 2]
```diff
@@ -82,13 +66,50 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 		},
 		Status: configv1.InfrastructureStatus{
 			InfrastructureName:   clusterID.InfraID,
-			Platform:             platform,
 			APIServerURL:         getAPIServerURL(installConfig.Config),
 			APIServerInternalURL: getInternalAPIServerURL(installConfig.Config),
 			EtcdDiscoveryDomain:  getEtcdDiscoveryDomain(installConfig.Config),
+			PlatformStatus:       &configv1.PlatformStatus{},
```
Comment (runcom) [inline pkg/asset/manifests/infrastructure.go]: The MCO was reading `Platform`- is it just enough to switch to `PlatformStatus`? cc @sgreene570

Comment (abhinavdahiya) [reply pkg/asset/manifests/infrastructure.go]: This is only set in new 4.2 clusters.

4.1 clusters do not have `.status.platformStatus` set.

Comment (runcom) [reply pkg/asset/manifests/infrastructure.go]: I know, so it's ok to switch to the new platformStatus in 4.2 and live with that as you're doing in this PR?

**Pull request merged**